lib/pure/hashes.nim

```diff
@@ -197,8 +197,29 @@ else:
 
 proc hash*(x: float): Hash {.inline.} =
   ## Efficient hashing of floats.
-  var y = x + 0.0 # for denormalization
-  result = hash(cast[ptr Hash](addr(y))[])
+  let y = x + 0.0 # for denormalization
+  when nimvm:
+    when defined(js):
+      # workaround a JS VM bug
+      # TODO see issue #16547
```

Comment:
```suggestion
      # workaround a JS VM bug: bug #16547
```
https://nim-lang.github.io/Nim/contributing.html
> Always refer to a GitHub issue using the following exact syntax: bug #1234 as shown above, so that it's consistent and easier to search or for tooling [...]